code/_onclick/ai_onclick.dm

```diff
@@ -216,7 +221,12 @@
 	if(isElectrified())
 		electrify(0, user, TRUE) // un-shock
 	else
-		electrify(-1, user, TRUE) // permanent shock
+		if(isAntag(user) || do_after(user, 3 SECONDS, target = src, allow_moving = TRUE))
+			electrify(-1, user, TRUE) // permanent shock + audio cue
+			playsound(loc, "sparks", 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+			return
+		electrify(-1, user, TRUE)// permanent shock + audio cue
+		playsound(loc, "sparks", 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```

Comment:
```suggestion
```

Reply:
This is required so that if the AI is malf, it can insta-shock doors (unless im missing something here).

Reply:
Since it's an || expression, the first (read: leftmost) expression that evaluates to true will cause the entire expression to evaluate as true and skip evaluating everything else (it "short-circuits"). This means that if the `isAntag()` call returns true, it won't call the `do_after` at all. If the `isAntag()` call returns *false*, though, it'll check the `do_after()` expression evaluates to true, which will cause the expected delay.

So, yeah. You should get rid of the call below, because as it stands, if you aren't an antag and you cancel the do-after, the door will immediately electrify.